src/transformStateWithClones.js

```
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let prevState = { ...state };
+
+  for (const actionObj of actions) {
+    const action = actionObj.type;
+    let nextState;
+
+    switch (action) {
+      case 'addProperties':
+        nextState = { ...prevState, ...actionObj.extraData };
+        break;
+
+      case 'removeProperties':
+        nextState = { ...prevState };
+
+        for (const key of actionObj.keysToRemove) {
+          delete nextState[key];
+        }
+        break;
+
+      case 'clear':
+        nextState = {};
+        break;
+
+      default:
+        continue;
```

There, we need to throw an error about an unsupported type of action or something similar to this, at your choice.

---

Fixed that in commit [43d3942]